designs/plugin-resolution.md

```diff
@@ -0,0 +1,88 @@
+# Plugin and project version resolution
+
```

Comment:
IMO we need to improve this doc a little. We might add a template for the proposals here such as; https://github.com/operator-framework/enhancements/blob/master/enhancements/template.md

Reply:
I was thinking that this probably shouldnt be merged, we should treat this as an issue nad close it when solved.

Creating a template for EPs sounds fine though.